content.go

```diff
@@ -76,25 +93,116 @@ func Resolve(ctx context.Context, target ReadOnlyTarget, ref string, opts Resolv
 		switch desc.MediaType {
 		case docker.MediaTypeManifestList, ocispec.MediaTypeImageIndex,
 			docker.MediaTypeManifest, ocispec.MediaTypeImageManifest:
-			// create a proxy to cache the fetched descriptor
-			store := cas.NewMemory()
-			err = store.Push(ctx, desc, rc)
+			// cache the fetched content
+			err = proxy.Cache.Push(ctx, desc, rc)
 			if err != nil {
 				return ocispec.Descriptor{}, err
 			}
-
-			proxy := cas.NewProxy(target, store)
+			// stop caching as SelectManifest may fetch a config blob
 			proxy.StopCaching = true
-			return selectPlatform(ctx, proxy, desc, opts.TargetPlatform)
+			return platform.SelectManifest(ctx, proxy, desc, opts.TargetPlatform)
 		default:
 			return ocispec.Descriptor{}, fmt.Errorf("%s: %s: %w", desc.Digest, desc.MediaType, errdef.ErrUnsupported)
 		}
 	}
 
-	desc, err := target.Resolve(ctx, ref)
+	desc, err := target.Resolve(ctx, reference)
 	if err != nil {
 		return ocispec.Descriptor{}, err
 	}
+	return platform.SelectManifest(ctx, target, desc, opts.TargetPlatform)
+}
+
+// Fetch fetches the content identified by the reference.
+func Fetch(ctx context.Context, target ReadOnlyTarget, reference string) (ocispec.Descriptor, io.ReadCloser, error) {
```

Comment:
I'm wondering if this is necessary. If so, it looks like we should also provide a platform option.

Reply:
OK let me refactor this.